src/main/kotlin/racingcar/model/Car.kt

```diff
@@ -0,0 +1,20 @@
+package racingcar.model
+
+data class Car(
+    val carName: String,
+    val position: Int = INITIAL_POSITION
+) {
+    fun move(generator: NumberGenerator): Car {
+        val randomValue = generator.generate()
+        return if (randomValue >= MOVE_FORWARD_THRESHOLD) {
+            copy(position = position + MOVE_DISTANCE)
+        } else this
+    }
```

Comment:
copy()로 불변성을 유지하신 설계도 좋지만,
private var position으로 바꾸면 매번 객체를 복사하지 않고 관리할 수 있어서 조금 더 효율적일 것 같습니다!

```suggestion
data class Car(
    val carName: String,
    private var position: Int = INITIAL_POSITION
) {
    fun move(generator: NumberGenerator){
        val randomValue = generator.generate()
        if (randomValue >= MOVE_FORWARD_THRESHOLD) {
            position++
        }
    }
    fun getPosition() : Int = position
```

또한 data class는 일반적으로 상태를 표현하는 목적이 강해서,
내부에서 전진 여부를 결정하는 행위 로직이 들어가면 약간 어색할 수도 있을 것 같습니다.

만약 행위 중심 객체로 보신다면 data class 대신 class로 변경하거나
외부에서 전진 여부를 판단하고 `move()`에서는 단순히 위치만 증가시키는 형태로 분리해도 좋을 것 같아요!

Reply:
저도 아무래도 구현하면서 data class로 행위 로직을 표현하기에는 약간 애매한 부분이 있다고 느꼈는데 다시 돌아보니 확실히 아쉽다고 느껴지네요, 
말씀 주신대로 일반 클래스로 변경하고  Position 속성은 private var로 바꾸는 방안이 제가 생각하기에도 더 좋은 것 같습니다 
코드 수정 제안까지 달아주시고 상세한 리뷰 정말 감사드립니다..!